lib/subtitle.rb

```diff
@@ -68,6 +73,66 @@ def transform(types, options = nil, target_lang = nil, src_lang = nil)
     @handler.transform_to(vals, src_lang, target_lang, output_dir)
   end
 
+  def generate_srt(outfile)
+    srt_helper = SrtHelper.new
+    srt_helper.parse_file(@cc_file, outfile)
+  end
+
+  def transcribe(options)
+    # From here call the "aws.rb"
+    # Get the transcribe output json file 
+    # Pass it to the generate_srt
+
+    aws = AwsEngine.new(options)
+    outfile = options[:outfile]
+    bucket = options[:bucket]
+    audio_lang = options[:audio_lang]
+    video_file = options[:video_file]
+    transcribe_json_file = nil
+    isurl = isURL?(video_file)
+    if(isurl == true)
+      output = aws.transcribe_uri(video_file, audio_lang, bucket)
+    else
+      output = aws.transcribe_file(video_file, audio_lang, bucket)
+    end
+    if output["status"].eql?("FAILED")
```

Comment:
If the status is failed, we should only do cleanup right below and not attempt parsing using srt_helper

Reply:
Need to raise error, so raised an error.